src/basic/chase.c

```diff
@@ -665,7 +668,7 @@ int chase(const char *path, const char *root, ChaseFlags flags, char **ret_path,
         if (ret_fd)
                 *ret_fd = TAKE_FD(pfd);
 
-        return r;
+        return trailing_slash ? O_DIRECTORY : r;
```

Comment:
No, this breaks existing behavior.
Currently, `chaseat()` returns 0 when
- the file does not exist and `CHASE_NONEXISTENT` flag is set,
- `CHASE_STEP` is specified and only a part of the path is resolved,

and returns 1 on other success cases.

At least, the behavior of CHASE_STEP should be kept.

Reply:
Okay thanks for the review.